boot/makebootable.go

```diff
@@ -167,6 +167,22 @@ func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet)
 		return fmt.Errorf("internal error: cannot find bootloader: %v", err)
 	}
 
+	if bootWith.RecoverySystemLabel == "" {
+		return fmt.Errorf("internal error: recovery system label unset")
+	}
+
+	// record which recovery system is to be used on the bootloader, note that
+	// this goes on the main bootloader, and not on the recovery system 
+	// bootloader, for example for grub bootloader, this env var is set on 
+	// the ubuntu-seed root grubenv, and not on the recovery system grubenv in 
+	// the systems/20200314/ subdir on ubuntu-seed
+	blVars := map[string]string{
+		"snapd_recovery_system": bootWith.RecoverySystemLabel,
+	}
+	if err := bl.SetBootVars(blVars); err != nil {
+		return fmt.Errorf("cannot set system environment: %v", err)
```

Comment:
s/system/recovery/

Reply:
relatedly, there's an error in makeBootable20RunMode that says:

"cannot set recovery system environment"

it should drop the "system" bit